src/SonarScanner.MSBuild.PreProcessor/HostInfo.cs

```diff
@@ -85,7 +85,9 @@ public record CloudHostInfo(string ServerUrl, string ApiBaseUrl, string Region)
     public static new CloudHostInfo FromProperties(ILogger logger, string sonarHostUrl, string sonarCloudUrl, string apiBaseUrl, string region)
     {
         region = region?.Trim().ToLower() ?? string.Empty;
-        if (KnownRegions.SingleOrDefault(x => x.Region == region) is { } knownRegion)
+        var regionFromUrl = region == string.Empty ? KnownRegions.FirstOrDefault(x => UrlEquals(x.ServerUrl, sonarHostUrl) || UrlEquals(x.ServerUrl, sonarCloudUrl)) : null;
```

Comment:
If it equals, it rather equal exactly once. Otherwise, we have a problem in `KnownRegions` values
```suggestion
        var regionFromUrl = region == string.Empty ? KnownRegions.SingleOrDefault(x => UrlEquals(x.ServerUrl, sonarHostUrl) || UrlEquals(x.ServerUrl, sonarCloudUrl)) : null;
```